### PR TITLE
On Fuchsia, always composite physical model layers.

### PR DESCRIPTION
**Pull request by j9brown**
The system compositor will draw the shadows itself.

### CODE REVIEW

**Comment by j9brown** [comment]
/cc @chinmaygarde 

We'll need this for Mozart v2.

**Review by chinmaygarde — APPROVED** [review]

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:io' show Platform;
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: This should be based on defaultTargetPlatform in foundation.dart, not on dart:io's Platform

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 2]
```diff
@@ -1326,6 +1327,9 @@ class RenderPhysicalModel extends _RenderCustomClip<RRect> {
   static final Paint _transparentPaint = new Paint()..color = const Color(0x00000000);
 
   @override
+  bool get alwaysNeedsCompositing => Platform.isFuchsia;
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: you should add a comment explaining why

**Comment by Hixie** [comment]
test?

File: packages/flutter/lib/src/rendering/proxy_box.dart (hunk 2 above)
Comment (j9brown) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: Hmm, perhaps I should have this ask SceneBuilder instead so the logic and reasoning is centralized.

**Comment by Hixie** [comment]
LGTM

**Pull request merged**